homeassistant/components/google_generative_ai_conversation/tts.py

```diff
@@ -0,0 +1,232 @@
+        self._attr_unique_id = f"{entry.entry_id}_tts"
+        self._attr_device_info = dr.DeviceInfo(
+            identifiers={(DOMAIN, entry.entry_id)},
+            name=entry.title,
+            manufacturer="Google",
+            model="Generative AI",
+            entry_type=dr.DeviceEntryType.SERVICE,
+        )
+        self._genai_client = entry.runtime_data
+        self._default_voice_id = self._supported_voices[0].voice_id
+
+    @callback
+    def async_get_supported_voices(self, language: str) -> list[Voice] | None:
+        """Return a list of supported voices for a language."""
+        return self._supported_voices
+
+    async def async_get_tts_audio(
+        self, message: str, language: str, options: dict[str, Any]
+    ) -> TtsAudioType:
+        """Load tts audio file from the engine."""
+        try:
+            response = self._genai_client.models.generate_content(
+                model=options.get(ATTR_MODEL, RECOMMENDED_TTS_MODEL),
+                contents=message,
+                config=types.GenerateContentConfig(
+                    response_modalities=["AUDIO"],
+                    speech_config=types.SpeechConfig(
+                        voice_config=types.VoiceConfig(
+                            prebuilt_voice_config=types.PrebuiltVoiceConfig(
+                                voice_name=options.get(
+                                    ATTR_VOICE, self._default_voice_id
+                                )
+                            )
+                        )
+                    ),
+                ),
+            )
+
+            data = response.candidates[0].content.parts[0].inline_data.data
+            mime_type = response.candidates[0].content.parts[0].inline_data.mime_type
+        except Exception as exc:
+            _LOGGER.warning(
+                "Error during processing of TTS request %s", exc, exc_info=True
+            )
+            raise HomeAssistantError(exc) from exc
+        return "wav", self._convert_to_wav(data, mime_type)
+
+    def _convert_to_wav(self, audio_data: bytes, mime_type: str) -> bytes:
+        """Generate a WAV file header for the given audio data and parameters.
+
+        Args:
+            audio_data: The raw audio data as a bytes object.
+            mime_type: Mime type of the audio data.
+
+        Returns:
+            A bytes object representing the WAV file header.
+
+        """
+        parameters = self._parse_audio_mime_type(mime_type)
+        bits_per_sample = parameters["bits_per_sample"]
+        sample_rate = parameters["rate"]
+        num_channels = 1
+        data_size = len(audio_data)
+        bytes_per_sample = bits_per_sample // 8
+        block_align = num_channels * bytes_per_sample
+        byte_rate = sample_rate * block_align
+        chunk_size = 36 + data_size  # 36 bytes for header fields before data chunk size
+
+        # http://soundfile.sapp.org/doc/WaveFormat/
+
+        header = struct.pack(
```

Comment:
Can you use the wave library here?

Something like the following should work

```python
wav_buffer = io.BytesIO()
with wave.open(wav_buffer, 'wb') as wf:
    wf.setnchannels(num_channels)
    wf.setsampwidth(bytes_per_sample)
    wf.setframerate(sample_rate)
    wf.writeframes(audio_data)

return wav_buffer.getvalue()
```

Reply:
Changed it to use the wave library and verified that it works properly